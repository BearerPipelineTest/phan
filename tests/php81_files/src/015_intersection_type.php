<?php // NOTE: This won't work properly in Phan 4, upgrade to Phan 5
function example(ArrayAccess&Countable $arr) {
    var_export($arr[0]);
    var_export(count($arr));
    // Not typed as Traversable
    foreach ($arr as $invalid) {
        var_dump($invalid);
    }
    var_dump(new $arr);  // TODO: Stop emitting PhanTypeExpectedObjectOrClassName?
    var_dump(clone $arr);
}
class Uncountable implements Countable {
    public function count(): int {
        return 0;
    }
}
example(new ArrayObject());
example(new Uncountable());
