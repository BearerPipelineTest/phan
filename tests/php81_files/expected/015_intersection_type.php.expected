%s:6 PhanTypeNoPropertiesForeach Class \ArrayAccess was passed to foreach, but it does not extend Traversable and doesn't have any declared properties. (This check excludes dynamic properties)
